streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -1199,6 +1199,9 @@ private void verifyMaxInFlightRequestPerConnection(final Object maxInFlightReque
 
         // no need to set group id for a restore consumer
         baseConsumerProps.remove(ConsumerConfig.GROUP_ID_CONFIG);
+        // no need to set instance id for a restore consumer
+        baseConsumerProps.remove(ConsumerConfig.GROUP_INSTANCE_ID_CONFIG);
```

Comment:
Side note: We actually have `NON_CONFIGURABLE_CONSUMER_DEFAULT_CONFIGS` (similar for producer and EOS). I am wondering, if we should add a similar pattern for the restore and global consumer (might be good enough as a follow up PR though; or at least create a ticket for now?). This include logging a WARNING if a user tries to set it?

Reply:
Yeah I think that's a good idea, can be done in another PR.